### ISSUE TITLE
Unable to resolve content item other than homepage

### ISSUE THREAD

**Issue by xtomas** [EtchUK/Etch.OrchardCore.ContextualEdit#13]
I've used your NuGet package in current version 1.4.0 in OrchardCore site and followed your instructions. Widget is displayed on homepage only.

So I downloaded your source code from github and referenced project to be able to debug.

The problem I've found is in the `ContextualEditService`, method `GetContentItemForRequestAsync`, line 42. The variable `contentItemId` is always `null`. And the reason is that `_contentHandleProvider` instance variable is `AliasPartContentHandleProvider` type instead of expected `AutorouteHandleProvider` type.

One solution that works for me is replace IContentHandleProvider with [IContentHandleManager](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.ContentManagement/ContentHandleManager.cs). It will use all content handle providers ordered by Order property and return first successfull result.

Another solution, that also works is to resolve `AutorouteHandleProvider` from content handle provider collection registered in dependency injection container. This solution will use only expected content handle provider. May there is another solution to resolve correct service from DI but currently I don't know how.

```csharp
public ContextualEditService(IEnumerable<IContentHandleProvider> contentHandleProviders, IContentManager contentManager, IHttpContextAccessor httpContextAccessor, ISiteService siteService)
{
    _contentHandleProvider = contentHandleProviders.OfType<AutorouteHandleProvider>().First();
    _contentManager = contentManager;
    _httpContextAccessor = httpContextAccessor;
    _siteService = siteService;
}
```

**Comment by peterkeating** [comment]
>  The variable contentItemId is always null. And the reason is that _contentHandleProvider instance variable is AliasPartContentHandleProvider type instead of expected AutorouteHandleProvider type.

@xtomas Is this because the content type doesn't have an autoroute part?

**Comment by xtomas** [comment]
@peterkeating The content type has AutoroutePart and AliasPart too. It's a long time I debugged it, but may be it had occured on content type marked as homepage. I'm using the solution above for almost 1 year without any problems. I will debug it again to retest if it still happens in current version with Orchard 1.5.0.

**Comment by peterkeating** [comment]
@xtomas We should catch the content item that is the homepage by looking at whether the path is empty or "/", in which case we grab the content item that is currently set as the homepage.

If you could debug it and let me know and hopefully we can work out why it wasn't working for you.

**Comment by PBMikeW** [comment]
I think this is related to the content handler used as per my pull request, if alias feature is turned on it does not get the contentitemid

**Comment by xtomas** [comment]
Finally I tried again to debug code. I tried all variants and here are the results:

![image](https://user-images.githubusercontent.com/10938220/224673126-e9c7dd75-ab68-4a4d-8128-ef1e34196980.png)

![image](https://user-images.githubusercontent.com/10938220/224673372-64618413-b82a-47f1-b33b-ab31b1ef4cde.png)

So the PR #15 prepared by @PBMikeW is functional solution for me.

**Comment by peterkeating** [comment]
Closing as @PBMikeW change has been included in 1.5.1. Thank you both!
